library/core/src/pin/unsafe_pinned.rs

```diff
@@ -86,11 +86,14 @@ impl<T: ?Sized> UnsafePinned<T> {
         ptr::from_mut(self) as *mut T
     }
 
-    /// Get read-only access to the contents of a shared `UnsafePinned`.
+    /// Get mutable access to the contents of a shared `UnsafePinned`.
     ///
-    /// Note that `&UnsafePinned<T>` is read-only if `&T` is read-only. This means that if there is
-    /// mutation of the `T`, future reads from the `*const T` returned here are UB! Use
-    /// [`UnsafeCell`] if you also need interior mutability.
+    /// This can be cast to a pointer of any kind.
+    /// Ensure that the access is unique (no active references, mutable or not)
+    /// when casting to `&mut T`, and ensure that there are no mutations
+    /// or mutable aliases going on when casting to `&T`.
```

Comment:
I'd suggest wording here along the lines of "...ensure that there are no ongoing mutations or live mutable references when casting to `&T`."

Reply:
FWIW this paragraph is copied verbatim from `UnsafeCell::get`. I was hoping we could just reuse it without spending a lot of time on editing.^^